internal/arenaskl/arena.go

```diff
@@ -76,15 +88,15 @@ func (a *Arena) alloc(size, align, overflow uint32) (uint32, uint32, error) {
 	}
 
 	// Pad the allocation with enough bytes to ensure the requested alignment.
-	padded := uint32(size) + align
+	padded := size + align - 1
 
 	newSize := a.n.Add(uint64(padded))
 	if int(newSize)+int(overflow) > len(a.buf) {
 		return 0, 0, ErrArenaFull
 	}
 
 	// Return the aligned offset.
-	offset := (uint32(newSize) - padded + align) & ^align
+	offset := (uint32(newSize) - size) & ^(align - 1)
```

Comment:
super nit: maybe throw in an invariants.Enabled guarded panic for align not being a power of 2?